### PR TITLE
Frontend + API Plan

### PR DESCRIPTION
**Pull request by stevenelleman**
## What 
1. Write up frontend plan at `app/src/components/README.md` 
2. Clean up directory (remove unnecessary files from create-react-app)

## How 
Present some semi-organized thoughts on how the frontend and API will be organized. While it makes sense in my mind and I can picture the frontend react application and API service, I suspect others won't. That's what review is for: please make it clear what's not clear! 

## Why 
For large code changes it's best to get feedback before any code is written. 

## Tasks 
- [x] Go through round of review, make sure others are onboard before jumping into coding.

## Notes to Reviewers

### CODE REVIEW

File: frontend/src/components/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+Client-Side: Apply patches to current state 
+
+Patches are a v1 concern. 
+
+## Routing 
+In the documents view I want to manage multiple RW (read/write) panels simultaneously. 
+
+This router will need to be applied only for the documents view (but should save state even when the documents view is left - i.e. the docs and state should be cached/saved).
+
+When a panel is selected a `focused` boolean will be set to `true`. While focused all requests will apply for that scope, as identified by its uuid. This means the router will only be applied to one scope at any given time.
```
Comment (Laserbear) [inline frontend/src/components/README.md:62]: How are we updating links? If i create a source linking text between two documents, that request can be sent through one scope, but needs to update both documents. Will we just poll for these updates when the view is switched again?

File: frontend/src/components/README.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+Give a UI options for all FIs, explicitly marking them includes the corresponding FI with the request. The cookie is then translated into the corresponding identity in some authn/authz step in the API. That way only the selected FI will be included in the responses.
+
+## API Routes 
+
+Silos: An authorization context must be included that includes a Faceted Identity (FI) of a silo member. All child routes must also include corresponding FI. 
+- `GET`-`v1/silos`: List silos of FIs.  
+- `GET`-`v1/silos/${id}`: Get silo details.
+- `POST`-`v1/silos/${id}`: Create silo.  
+- `PUT`-`v1/silos/${id}`: Edit silo. 
+- `DELETE`-`v1/silos/${id}`: Soft-delete silo. Unless explicitly soft-deleted, scopes are opened up to the public.
```
[the remark this answers is not in the document]
Comment (nvmandi7) [reply frontend/src/components/README.md:76]: I guess that seems reasonable as long as
1. it's made forwardly clear to the users
and 2. they have the option to delete things permanently
People value privacy

File: backend/mock/mock.js [hunk 3]
```diff
@@ -0,0 +1 @@
+// Placeholder file to create directory structure
```
Comment (nvmandi7) [inline backend/mock/mock.js:1]: newline here too

File: frontend/src/components/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevenelleman) [reply frontend/src/components/README.md:76]: Agreed.

File: frontend/src/components/README.md (hunk 1 above)
Comment (stevenelleman) [reply frontend/src/components/README.md:62]: Polling is the first thing that comes to mind, but there would be some complexity/overhead for all the panels to be polling constantly. A cheaper/simpler alternative would be triggering an update for each impacted panel.

Comment (stevenelleman) [reply frontend/src/components/README.md:62]: This is a good question. I'll put this down somewhere.

**Pull request merged**